### DISCUSSION TITLE
Cookie-based session

### DISCUSSION THREAD

**Discussion by BarneyKY** [tokio-rs/axum#3029] [Q&A] [answered]
### Summary

Hello everyone, i like axum so much that i cannot wait to use it, but there is one thing makes me confused, i cannot find a signed `Cookie-based` crate for session, just like `actix-session` which can choose the cookie-based sessionstore.

Is there any recommendation?

Or if the cookie-based plan is not recommeded ,what is the reason for it?

Thanks!

### axum version

0.7.7

**Comment by jplatte** [comment, marked as answer]
There are many existing crates that help with this. `axum-extra` offers multiple extractors in the [`cookie`](https://docs.rs/axum-extra/latest/axum_extra/extract/cookie/index.html) module and there's also different third-party cookie- / session-related projects linked in [`ECOSYSTEM.md`](https://github.com/tokio-rs/axum/blob/main/ECOSYSTEM.md).

**Reply by BarneyKY** [reply]
Got it!

Thanks a lot!
